src/test/java/io/github/jancieslak/tba/ScreeningServiceTest.java

```diff
@@ -0,0 +1,173 @@
+        screeningService = new ScreeningService(mockScreeningRepository, mockScreeningRoomRepository, mockSeatRepository);
+    }
+
+    @Test
+    public void getScreenInIntervalTest() {
+        // Wrong time
+        var todayDate = LocalDate.now();
+        var badBeforeTime = LocalTime.of(10, 0);
+        var badAfterTime = LocalTime.of(9, 0);
+        var badInput = new GetScreeningsRequestModel(LocalDateTime.of(todayDate, badBeforeTime), LocalDateTime.of(todayDate, badAfterTime));
+        assertNull(screeningService.getScreeningsInInterval(badInput));
+
+        // Wrong date
+        var anotherDate = LocalDate.MIN;
+        var beforeTime = LocalTime.of(8, 0);
+        var afterTime = LocalTime.of(19, 0);
+        var badInput2 = new GetScreeningsRequestModel(LocalDateTime.of(todayDate, beforeTime), LocalDateTime.of(anotherDate, afterTime));
+        assertNull(screeningService.getScreeningsInInterval(badInput2));
+
+        var batmanMovie = Movie.builder().title("Batman").build();
+        var jawsScreening = Screening.builder()
+                .movie(Movie.builder().title("Jaws").build())
+                .fromDateTime(LocalDateTime.of(todayDate, LocalTime.of(15, 0)))
+                .toDateTime(LocalDateTime.of(todayDate, LocalTime.of(17, 0)))
+                .build();
+
+        var batmanScreening1 = Screening.builder()
+                .movie(batmanMovie)
+                .fromDateTime(LocalDateTime.of(todayDate, LocalTime.of(19, 0)))
+                .toDateTime(LocalDateTime.of(todayDate, LocalTime.of(21, 0)))
+                .build();
+
+        var batmanScreening2 = Screening.builder()
+                .movie(batmanMovie)
+                .fromDateTime(LocalDateTime.of(todayDate, LocalTime.of(8, 0)))
+                .toDateTime(LocalDateTime.of(todayDate, LocalTime.of(12, 0)))
+                .build();
+
+        when(mockScreeningRepository.findByDateTimeInterval(any(), any()))
+                .thenReturn(List.of(
+                        jawsScreening,
+                        batmanScreening1,
+                        batmanScreening2
+                ));
+
+        var input = new GetScreeningsRequestModel(LocalDateTime.of(todayDate, beforeTime), LocalDateTime.of(todayDate, afterTime));
+        var result = screeningService.getScreeningsInInterval(input);
+
+        assertEquals(3, result.size());
+
+        // #1 Item
+        assertEquals("Batman", result.get(0).getTitle());
+        assertEquals(LocalDateTime.of(todayDate, LocalTime.of(8, 0)), result.get(0).getFrom());
+        // #2 Item
+        assertEquals("Batman", result.get(1).getTitle());
+        assertEquals(LocalDateTime.of(todayDate, LocalTime.of(19, 0)), result.get(1).getFrom());
+        // #3 Item
+        assertEquals("Jaws", result.get(2).getTitle());
+        assertEquals(LocalDateTime.of(todayDate, LocalTime.of(15, 0)), result.get(2).getFrom());
+    }
+
+    @Test
+    public void getScreeningSeatsTest() {
```

Comment:
czy te testy nie są za dużę?

Reply:
Tutaj pomyślałem, że jeszcze są w dopuszczalnym rozmiarze, ale gdybym pomyślał inaczej to bym przeniósł inicjalizację danych wyżej jak w przypadku ReservationServiceTest